fsm/leave_add_csync.go

```diff
@@ -207,6 +204,21 @@ func LeaveAddCsyncAction(z *music.Zone) bool {
 	}
 
 	updater := music.GetUpdater(leavingSigner.Method)
+	err, csyncrrs := updater.FetchRRset(leavingSigner, z.Name, z.Name, dns.TypeCSYNC)
+	if err != nil {
+		err, _ = z.SetStopReason(fmt.Sprintf("Unable to fetch CSYNC RRset from %s: %v", leavingSigner.Name, err))
+		return false
```

Comment:
IMO this function should return the error instead of a bool. The calling function should make the decision what to do with the error and maybe handle it differently based on what error is returned.

Reply:
Valid point, this needs to be implemented through the entire code base.